### PR TITLE
feat: updated Instructor-XL, GPU and deploy configs

### PR DESCRIPTION
**Pull request by gphorvath**

### CODE REVIEW

File: .python-version
Comment (gphorvath) [inline .python-version:1]: Think we need to gitignore this

File: CODEOWNERS
Comment (gphorvath) [inline CODEOWNERS:1]: I think CODEOWNERS should be done through a separate PR.

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Pipeline and Project Files
+/.github/ @gerred @YrrepNoj @gphorvath
+/.gitignore @gerred @YrrepNoj @gphorvath
+/Makefile @gerred @YrrepNoj @gphorvath
+/pyproject.toml @gerred @YrrepNoj @gphorvath
+/requirements-dev.txt @gerred @YrrepNoj @gphorvath
+/requirements.txt @gerred @YrrepNoj @gphorvath
+
+# Additional privileged files
+/CODEOWNERS @jeff-mccoy @mikevanhemert @austenbryan @gerred
```
Comment (gphorvath) [inline CODEOWNERS:23]: Remove Mike V (he is departing soon).

File: CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+/zarf-config.yaml @gerred @YrrepNoj @gphorvath
+/zarf.yaml @gerred @YrrepNoj @gphorvath
+
+# Pipeline and Project Files
+/.github/ @gerred @YrrepNoj @gphorvath
+/.gitignore @gerred @YrrepNoj @gphorvath
+/Makefile @gerred @YrrepNoj @gphorvath
+/pyproject.toml @gerred @YrrepNoj @gphorvath
+/requirements-dev.txt @gerred @YrrepNoj @gphorvath
+/requirements.txt @gerred @YrrepNoj @gphorvath
```
Comment (gphorvath) [inline CODEOWNERS:20]: Change all of these to `@defenseunicorns/leapfrogai` so we don't have as much friction getting PRs sorted out.

File: Dockerfile [hunk 3]
```diff
@@ -1,14 +1,35 @@
...
 
-FROM python:3.11-slim
 COPY requirements.txt .
+RUN pip install -r requirements.txt
 
-RUN pip install -r requirements.txt --user
+# download model
+ARG REPO_ID="hkunlp/instructor-xl"
+ARG REVISION="ce48b213095e647a6c3536364b9fa00daf57f436"
+COPY scripts/ scripts/
```
Comment (YrrepNoj) [inline Dockerfile:18]: Instead of copying the entire scripts directory, only copy the one script that we need.

File: zarf.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+kind: ZarfPackageConfig
+metadata:
+  name: "###ZARF_PKG_TMPL_NAME###"
+  version: "###ZARF_PKG_TMPL_IMAGE_VERSION###"
```
Comment (YrrepNoj) [inline zarf.yaml:4]: Note to myself; we should be using a tool like [release-please](https://github.com/googleapis/release-please) to set the version of the package instead of relying on image version of the model-backend.

File: tests/test.py
Comment (YrrepNoj) [inline tests/test.py:1]: It might be worth us pairing for a couple hours to see if we can run this test as part of a workflow. 

I think we can get away with just running the model locally on the size of runners we have access to on this repo.

File: pyproject.toml [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
-name = "leapfrogai-backend-instructor-xl"
-version = "0.4.0"
+name = "leapfrogai-backend-embeddings"
+version = "0.1.0"
+
+description = "Embeddings backend for LeapfrogAI"
+authors = [{ name = "LeapfrogAI Authors", email = "ai@defenseunicorns.com" }]
+license = { file = "LICENSE" }
+readme = "README.md"
+requires-python = ">=3.11.4, <3.12"
```
Comment (YrrepNoj) [inline pyproject.toml:9]: ping to 3.11.6 for now

File: Makefile [hunk 6]
```diff
@@ -18,15 +25,23 @@ build-requirements:
...
-	git clone https://huggingface.co/hkunlp/instructor-xl model & \
-	docker build -t ghcr.io/defenseunicorns/leapfrogai-backend-instructor-xl:${VERSION} .
+docker-build:
+	docker build -t ghcr.io/defenseunicorns/leapfrogai/text-embeddings:${VERSION}-${ARCH} --build-arg ARCH=${ARCH} .
+
+docker-run:
+	docker run -d -p 50051:50051 ghcr.io/defenseunicorns/leapfrogai/text-embeddings:${VERSION}-${ARCH}
+
+docker-run-gpu:
+	docker run --gpus device=0 -e GPU_ENABLED=true -d -p 50051:50051 ghcr.io/defenseunicorns/leapfrogai/text-embeddings:${VERSION}-${ARCH}
```
Comment (YrrepNoj) [inline Makefile:44]: nit; would be nice to varbalize the device value, but not something i'd block this PR over.

**Review by YrrepNoj — COMMENTED** [review]
misc comments and thoughts <3

File: zarf.yaml (hunk 4 above)
Comment (justinthelaw) [reply zarf.yaml:4]: Yes, will resolve in a diff PR and branch!

File: tests/test.py
Comment (justinthelaw) [reply tests/test.py:1]: Yes, will resolve in a diff PR and branch!

**Review by YrrepNoj — APPROVED** [review]
